builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -553,11 +553,13 @@ func resourceArmVirtualMachineCreate(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
-	osProfile, err := expandAzureRmVirtualMachineOsProfile(d)
-	if err != nil {
-		return err
+	if _, ok := d.GetOk("os_profile"); ok {
+		osProfile, err := expandAzureRmVirtualMachineOsProfile(d)
+		if err != nil {
+			return err
+		}
+		properties.OsProfile = osProfile
 	}
```

Comment:
@pearcec you can push additional commits to the branch and it'll update the PR - don't worry about squashing commits down, we can do that when we merge the PR :)